### ISSUE TITLE
Active drag data is not set on dragstart

### ISSUE THREAD

**Issue by pleku** [vaadin/vaadin-grid-flow#661]
When dragging rows from Grid/TreeGrid to another component by using the generic DnD add-on for Flow, one would expect to be able to get the active drag data, which is set by Grid/TreeGrid, from the `DropEvent` in another component.
https://github.com/vaadin/flow/blob/master/flow-dnd/src/main/java/com/vaadin/flow/component/dnd/DropEvent.java#L81

This is done in the generic DnD add-on for Flow by
1) setting the active drag source component (which both grid & tree grid should do)
https://github.com/vaadin/flow/blob/master/flow-dnd/src/main/java/com/vaadin/flow/component/dnd/DragSource.java#L174 
2) setting the drag data for the component, which should be the rows, in the drag start https://github.com/vaadin/flow/blob/master/flow-dnd/src/main/java/com/vaadin/flow/component/dnd/DragSource.java#L233

What makes this now difficult is that the constant (magic string) used for storing the drag data is not yet available for Grid because it cannot depend on the add-on. So using the add-on together with Grid/TreeGrid is very difficult without that.

As a workaround, one can use the examples from above to set the both 1) 2), with the following lines in you listener for `GridDragStart`:
```java
DragSource.configure(myGrid).setDragData(event.getDraggedItems());
event.getSource().getUI().get().getInternals().setActiveDragSourceComponent(event.getSource());
``` 
Then respectively in the listener for `GridDragEnd` you want to clear at the active drag source with
```java
event.getSource().getUI().get().getInternals().setActiveDragSourceComponent(null);
```

**Comment by mperktold** [comment]
It seems you fixed the issue you described in d1b8493d3a626d2399d7c87b19f289f392c8a114, so dragging grid items to other components now works as expected.

However, there is still an issue with the opposite direction, namely dragging data from other components into a Grid.
The problem is that the `GridDropEvent` doesn't have a reference to the drag source component and so you cannot access drag data inside your `GridDropEvent` listener.

This is easy to fix though, by doing it just like `DropEvent` does: Get the drag source component from the `UI` inside the onstructor and assign it to a class member, and provide getters both for the component and the data attached to it.

**Comment by pleku** [comment]
Right, the original issue was addressed, so basically what would be remaining is to add a 
```java
public Optional<Component> getDragSourceComponent() {  }
```
or
```java
public <S extends Component> Optional<S> getDragSourceComponent(Class<S> type) { }
```
where the latter could only return a drag source component of correct type if it matches the given type.

But this is actually an enhancement to the current API instead of a bug.
